custom-matchers/contracts/AssetMatcherWETH.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.7.6;
+pragma abicoder v2;
+
+import "@rarible/exchange-interfaces/contracts/IAssetMatcher.sol";
+import "@rarible/exchange-interfaces/contracts/IWETH.sol";
+
+/*
+ * Custom matcher for WETH_UNWRAP
+ */
+contract AssetMatcherWETH is IAssetMatcher {
+
+    bytes constant EMPTY = "";
+    string constant nameWETH = "Wrapped Ether";
+
+    function matchAssets(LibAsset.AssetType memory leftAssetType, LibAsset.AssetType memory rightAssetType) public view override returns (LibAsset.AssetType memory) {
+        bytes4 resultAssetClass;
+        bytes memory resultData;
+        if (
+            (leftAssetType.assetClass == LibAsset.ERC20_ASSET_CLASS) &&
+            (rightAssetType.assetClass == LibAsset.WETH_UNWRAP)
+        ) {
+            resultAssetClass = rightAssetType.assetClass;
+            resultData = leftAssetType.data;
+        } else if (
+            (leftAssetType.assetClass == LibAsset.WETH_UNWRAP) &&
+            (rightAssetType.assetClass == LibAsset.ERC20_ASSET_CLASS)
+        ) {
+            resultAssetClass = leftAssetType.assetClass;
+            resultData = rightAssetType.data;
+        } else {
+            return LibAsset.AssetType(0, EMPTY);
+        }
+        (address token) = abi.decode(resultData, (address));
+        try IWETH(token).name() returns (string memory name) {
+            if (keccak256(bytes(name)) == keccak256(bytes(nameWETH))) {
```

Comment:
Lets think about it 
Now I don`t understand how to store  token address in WETH_UNWRAP type
smthing like that:
bytes4 constant public WETH_UNWRAP_ADDRESS = bytes4(keccak256(abi.encode(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2))); //0x23a65993
where 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2 - is addr of WETH on mainnet

but WETH_UNWRAP type should be different for different test nets? because for different nets WETH will have different addresses after deploy

Reply:
agree, done